picard/script.py

```diff
@@ -1280,11 +1284,12 @@ def func_datetime(parser, format=None):
     try:
         return datetime.datetime.now(tz=local_tz).strftime(format)
     except ValueError:
+        function = parser._function_stack.get()
         raise ScriptRuntimeError(
             "Unsupported format code",
-            parser._function_stack.get(),
-            parser._y,
-            parser._x
+            function.function,
+            function.line,
+            function.column
```

Comment:
`function` is the tuple `FunctionStackItem`, right?
So we could just pass it to `ScriptRuntimeError`:

```python
stackitem = = parser._function_stack.get()
raise ScriptRuntimeError(
     "Unsupported format code",
     stackitem
)
```

Which makes me think we could perhaps make FunctionStackItem a class rather than a named tuple:

```python
class StackItem:
   def __init__(self, line, column):
       self.line = line
       self.column = column

class FunctionStackItem(StackItem):
     def __init__(self, line, column, name):
        super().__init__(line, column)
        self.name = name
```

So we could have something more consistent between all Script exceptions:

```python
raise ScriptRuntimeError("message", functionstackitem)
raise ScriptUnknownFunction("message", stackitem)
raise ScriptEndOfFile("message", stackitem)
raise ScriptSyntaxError("message", stackitem)
```